ndarray/src/main/java/org/tensorflow/ndarray/index/TensorIndex.java

```diff
@@ -12,29 +12,39 @@
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
- =======================================================================
+ ==============================================================================
  */
 package org.tensorflow.ndarray.index;
 
-import org.tensorflow.ndarray.impl.dimension.Dimension;
+public interface TensorIndex extends Index{
```

Comment:
I like that better.  For now, I've implemented it as a `tensorSupport` method, since the other methods have useful defaults and I don't like "override a method if it doesn't work", it's too easy to miss when adding indices.  The javadocs of those methods mention tensors too.  Do you think it would be better to call it non-DimensionalSpace or non-NDArray usage, instead of Tensor usage?